scripts/validate-accounts-hook.sh

```
@@ -0,0 +1,59 @@
+# Check if any destination config files have been modified
+dest_config_pattern="src/configurations/destinations/.*/db-config\.json"
+account_config_pattern="src/configurations/destinations/.*/accounts/.*/db-config\.json"
+
+# Extract unique destination names from changed files
+destinations=()
+
+for file in $changed_files; do
+  if [[ $file =~ src/configurations/destinations/([^/]+)/db-config\.json ]]; then
+    # Main destination config file changed
+    dest="${BASH_REMATCH[1]}"
+    if [[ ! " ${destinations[@]} " =~ " ${dest} " ]]; then
```

Shellcheck (0.10.0)

[error] Arrays implicitly concatenate in [[ ]]. Use a loop (or explicit * instead of @).

(SC2199)